lambdas/passwordRenewal/src/main/java/uk/gov/di/ipv/cri/drivingpermit/event/handler/PasswordRenewalHandler.java

```diff
@@ -0,0 +1,238 @@
+            SecretsManagerClient secretsManagerClient,
+            ChangePasswordService changePasswordService,
+            TokenRequestService tokenRequestService,
+            EventProbe eventProbe) {
+        this.secretsManagerClient = secretsManagerClient;
+        this.changePasswordService = changePasswordService;
+        this.tokenRequestService = tokenRequestService;
+        this.eventProbe = eventProbe;
+    }
+
+    @Logging
+    @Metrics(captureColdStart = true)
+    public Void handleRequest(SecretsManagerRotationEvent input, Context context) {
+        LOGGER.info("step {} secretId: {}", input.getStep(), input.getSecretId());
+
+        SecretsManagerRotationStep.of(input.getStep())
+                .ifPresent(
+                        step -> {
+                            try {
+                                if (step == SecretsManagerRotationStep.CREATE_SECRET) {
+                                    /*First step is for the new password to be generated, which will be labelled as 'AWSPENDING'
+                                     * and stored in the Secrets Manager as the pending new password until it's been verified by DVLA*/
+                                    LOGGER.info("Generating a new password");
+                                    String newPassword = generateNewPassword();
+
+                                    if (null
+                                            == getValue(
+                                                    secretsManagerClient, input.getSecretId())) {
+                                        PutSecretValueRequest secretRequest =
+                                                PutSecretValueRequest.builder()
+                                                        .secretId(input.getSecretId())
+                                                        .secretString(newPassword)
+                                                        .versionStages("AWSPENDING")
+                                                        .build();
+                                        secretsManagerClient.putSecretValue(secretRequest);
+                                    }
+                                    LOGGER.info("CREATE_SECRET step is complete");
+
+                                } else if (step == SecretsManagerRotationStep.SET_SECRET) {
+                                    /*Second step is to retrieve the AWS Pending Password and to send it to DVLA*/
+                                    String password = getValue(secretsManagerClient, DVLA_PASSWORD);
+                                    callDVLAApi(password);
+                                    LOGGER.info("SET_SECRET step is complete");
+
+                                } else if (step == SecretsManagerRotationStep.TEST_SECRET) {
+                                    /*Third step is to verify that the secret has been set in DVLA*/
+                                    String password = getValue(secretsManagerClient, DVLA_PASSWORD);
```

Comment:
Can we add a log here to note we have entered the test secret step.
Also can we leave a comment explaining we are requesting a token here only to ensure the new password works and we are not replacing the token in the DB. Additionally calling out that requesting a new token does not invalidate the existing one